### PR TITLE
Vulkan: add draw triangles

### PR DESCRIPTION
**Pull request by jaebaek**
Add tests/cases/draw_triangle_list.amber to test drawing simple
triangles and update engine to conduct draw based on parameters of
draw array command.

### CODE REVIEW

File: tests/cases/draw_triangle_list.amber [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+# limitations under the License.
+
+[vertex shader]
+#version 430
+
+layout(location = 0) in vec4 position;
+layout(location = 1) in vec4 vert_color;
+layout(location = 0) out vec4 frag_color;
+
+void
```
Comment (dj2) [inline tests/cases/draw_triangle_list.amber]: Let's just do `void main() {` to make this 2 lines shorter.

File: tests/cases/draw_triangle_list.amber [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  frag_color = vert_color;
+}
+
+[fragment shader]
+#version 430
+
+layout(location = 0) in vec4 frag_color;
+layout(location = 0) out vec4 final_color;
+
+void
```
Comment (dj2) [inline tests/cases/draw_triangle_list.amber]: ditto

**Comment by dj2** [comment]
Is there an associated bug for this CL?

File: tests/cases/draw_triangle_list.amber (hunk 1 above)
Comment (jaebaek) [reply tests/cases/draw_triangle_list.amber]: done

File: tests/cases/draw_triangle_list.amber (hunk 2 above)
Comment (jaebaek) [reply tests/cases/draw_triangle_list.amber]: done

**Comment by jaebaek** [comment]
PTAL

File: src/vulkan/engine_vulkan.cc [hunk 3]
```diff
@@ -44,7 +45,7 @@ VkShaderStageFlagBits ToVkShaderStage(ShaderType type) {
       return VK_SHADER_STAGE_COMPUTE_BIT;
   }
 
-  // Unreachable
+  assert(false && "Vulkan::Unknown shader stage");
```
Comment (dneto0) [inline src/vulkan/engine_vulkan.cc]: asserts only fire on debug builds. Should we have a project-wide macro or function to express the unreachable condition even in release builds?
Question for @dj2 as well.

File: tests/cases/draw_triangle_list.amber [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+#         R8   G8     R8  G8  B8
+           0 -128      0 255   0
+         127  127      0 255   0
+           0  127      0 255   0
+
+           0 -128      0 255   0
+         127  127      0 255   0
+         127 -128      0 255   0
+
+#          Blue for quater frame
```
Comment (dneto0) [inline tests/cases/draw_triangle_list.amber]: "quarter"

File: src/vulkan/engine_vulkan.cc (hunk 3 above)
Comment (dj2) [reply src/vulkan/engine_vulkan.cc]: There is an open bug to remove the asserts and return Results. Just have to do the refactoring.

File: tests/cases/draw_triangle_list.amber (hunk 4 above)
Comment (jaebaek) [reply tests/cases/draw_triangle_list.amber]: Done

**Pull request merged**